scripts/zones/Southern_San_dOria/npcs/Varchet.lua

```diff
@@ -77,19 +75,18 @@ function onEventFinish(player,csid,option)
     local exitTheGambler = player:getQuestStatus(SANDORIA,EXIT_THE_GAMBLER);
     local npc = player:getEventTarget();
 
-    if (csid == 519) then
-          if (option == GAME_WON) then
-            npc.giveGil(player, 10, {rate=false});
-        if (exitTheGambler == QUEST_ACCEPTED) then
-          player:completeQuest(SANDORIA,EXIT_THE_GAMBLER);
-          player:showText(npc,7629);
-        end
-          elseif (option == GAME_TIE) then
-            npc.giveGil(player, 5, {rate=false});
-          player:showText(npc,7622);
-          else
-            player:showText(npc,7620);
-          end
+    if (csid == 0x207) then
+        if (option == GAME_WON) then
+            player:addGil(10);
+            if (exitTheGambler == QUEST_ACCEPTED) then
+                player:completeQuest(SANDORIA,EXIT_THE_GAMBLER);
+                player:showText(npc,VARCHET_KEEP_PROMISE);
+            end
+        elseif (option == GAME_TIE) then
+            player:addGil(5);
+            -- player:showText(npc,7622); -- FIXME: What is he supposed to say on tie? Using this hard-coded ID softlocks.
```

Comment:
I know where his text ids are, but none of them seem to indicate that he tied the player, but I'll look again.